sequence-model-generator/modules/sequence-model-generator-ls-extension/src/test/java/io/ballerina/sequencemodelgenerator/ls/extension/ModelGeneratorTest.java

```diff
@@ -0,0 +1,148 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Test class for sequence model generator service.
+ *
+ * @since 2201.9.0
+ */
+public class ModelGeneratorTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ModelGeneratorTest.class);
+    private static final Path RES_DIR = Paths.get("src/test/resources/").toAbsolutePath();
+    private static final Path CONFIG_DIR = RES_DIR.resolve("config");
+    private static final Path SOURCE_DIR = RES_DIR.resolve("source");
+
+    private final Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+    private Endpoint serviceEndpoint;
+    private BallerinaLanguageServer languageServer;
+    private static final String SEQUENCE_DESIGN_SERVICE = "sequenceModelGeneratorService/getSequenceDiagramModel";
+
+    @BeforeClass
+    public void init() {
+        this.languageServer = new BallerinaLanguageServer();
+        TestUtil.LanguageServerBuilder builder = TestUtil.newLanguageServer()
+                .withLanguageServer(languageServer);
+        this.serviceEndpoint = builder.build();
+    }
+
+    @Test(dataProvider = "model-data-provider")
+    public void test(Path config) throws IOException {
+        Path configJsonPath = CONFIG_DIR.resolve(config);
+        TestConfig testConfig = gson.fromJson(Files.newBufferedReader(configJsonPath), TestConfig.class);
+        JsonObject responseJson = getResponse(testConfig.source(), testConfig.start(), testConfig.end());
+        Assert.assertNotNull(responseJson);
+        Diagram jsonModel = gson.fromJson(responseJson.getAsJsonObject("sequenceDiagram"), Diagram.class);
+
+        // Assert only the file name since the absolute path may vary depending on the machine
+        LineRange location = jsonModel.location();
+        String balFileName = Path.of(location.fileName()).getFileName().toString();
+        LineRange testConfigLocation = testConfig.diagram().location();
+        boolean fileNameEquality = testConfigLocation != null && balFileName.equals(testConfigLocation.fileName());
+        Diagram modifiedDiagram = new Diagram(
+                LineRange.from(balFileName, location.startLine(), location.endLine()), jsonModel.participants());
+
+        boolean flowEquality = modifiedDiagram.equals(testConfig.diagram());
+        if (!fileNameEquality || !flowEquality) {
+            TestConfig updatedTestConfig = new TestConfig(testConfig.source(), testConfig.description(),
+                    testConfig.start(), testConfig.end(), modifiedDiagram);
+            updateConfig(configJsonPath, updatedTestConfig);
+            Assert.fail(String.format("Failed test: '%s' (%s)", testConfig.description(), configJsonPath));
+        }
+    }
+
+    @DataProvider(name = "model-data-provider")
+    public Object[] getConfigsList() {
+//        return new Object[]{Path.of("transform_with_error.json")};
```

Comment:
```suggestion
```

Reply:
Similar to `updateConfig`, this was done to aid the developer in running isolated tests. A similar case can be found here: https://github.com/ballerina-platform/ballerina-dev-tools/blob/308a8be2c69012e1813e40719ffb586097e87794/worker-model-generator/modules/worker-model-generator-ls-extension/src/test/java/io/ballerina/workermodelgenerator/extension/ModelGeneratorTest.java#L102